pkg/cmd/roachtest/operations/add_index.go

```diff
@@ -28,11 +29,17 @@ func (cl *cleanupAddedIndex) Cleanup(
 	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
 	defer conn.Close()
 
+	if cl.locked {
+		setSchemaLocked(ctx, o, conn, cl.db, cl.table, false /* lock */)
```

Comment:
We can add the lock=true code as a differ after this line.
`defer setSchemaLocked(ctx, o, conn, cl.db, cl.table, true /* lock */)`